src/components/implementation/fork_tests/voter/voter.c

```diff
@@ -0,0 +1,290 @@
+#include <cos_component.h>
+#include <print.h>
+#include <quarantine.h>
+#include <cbuf.h>
+#include <sched.h>
+#include <voter.h>
+#include <periodic_wake.h>
+
+struct replica_info {
+	spdid_t spdid;
+	cbuf_t read_buffer;
+	cbuf_t write_buffer;
+	void *buf_read;
+	void *buf_write;
+	unsigned short int thread_id;
+	int blocked;
+};
+
+struct nmod_comp {
+	struct replica_info *replicas;
+	int nreplicas;
+};
+
+struct channel {
+	struct nmod_comp *A, *B;	// come up with better terms, they're symmetric
+	int timer; // is this an int, figure it out later
+	int A2B_data, have_A2B_data;	// make these arrays
+	int B2A_data, have_B2A_data;
+};
+
+/* number of components and component array */
+int n_comps = 2;
```

Comment:
Yes it should. I kind of did this, then realized why it was wrong, but didn't fix it, and now it happens in three places.